coffee/base.py

```diff
@@ -967,3 +1013,26 @@ def __eq__(self, other):
 DEC = Access("DEC")
 IMUL = Access("IMUL")
 IDIV = Access("IDIV")
+Access._modes = [READ, WRITE, RW, INC, DEC, IMUL, IDIV]
+
+
+# Scope of a declaration ##
+
+class Scope(object):
+
+    """An ``EXTERNAL`` scope means the /Decl/ is an argument of a kernel (i.e.,
+    when it appears in the list of declarations of a /FunDecl/ object). Otherwise,
+    a ``LOCAL`` scope indicates the /Decl/ is within the body of a kernel."""
+
+    def __init__(self, scope):
+        self._scope = scope
+
+    def __eq__(self, other):
+        return self._scope == other._scope
+
+    def __str__(self):
+        return self._scope
+
+LOCAL = Scope("LOCAL")
+EXTERNAL = Scope("EXTERNAL")
+Scope._scopes = [LOCAL, EXTERNAL]
```

Comment:
Don't do this, rather only allow people to build "LOCAL", or "EXTERNAL" scope objects.